scripts/Muon_Analysis.py

```diff
@@ -9,12 +9,17 @@
 from qtpy import QtCore
 from Muon.GUI.Common.usage_report import report_interface_startup
 
-
 Name = "Muon_Analysis_2"
 
 if 'muon_analysis' in globals():
     muon_analysis = globals()['muon_analysis']
-    if not muon_analysis.isHidden():
+    # If the object is deleted in the C++ side it can still exist in the
+    # python globals list. The try catch block below checks for this.
+    try:
+        is_hidden = muon_analysis.isHidden()
+    except RuntimeError:
+        is_hidden = True
```

Comment:
If it has been deleted on the C++ side, does it also need to be deleted from the `globals` list or is it fine to leave it there and just keep doing the check?

Reply:
Ideally it should be removed from the globals list as well. This is to correct an issue where when you closed the GUI the wrapped objects were deleted but it continued to exist in the global scope here but threw errors when you attempted to access anything. I am uncertain how to reliably remove it from this list in this case so for now added in the check.